app/Http/Controllers/UserAddressController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Repositories\UserAddressRepositoryEloquent;
+use App\Http\Requests\CreateUserAddressRequest;
+use App\Services\UserAddressService;
+use App\Http\Requests\UpdateUserAddressRequest;
+use App\Http\Resources\UserAddressResource;
+
+class UserAddressController extends Controller
+{
+    protected $userAddressRepository;
+
+    public function __construct(UserAddressRepositoryEloquent $userAddressRepository)
+    {
+        $this->userAddressRepository = $userAddressRepository;
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $userAddres = $this->userAddressRepository->all();
+        return UserAddressResource::collection($userAddres);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateUserAddressRequest $request, UserAddressService $userAddresService)
+    {
+        return new UserAddressResource($this->userAddressRepository->create($userAddresService->create($request)));
```

Comment:
gimana kalau dibuat jadi seperti ini ?
$data = $userAddresService->create($request);
return new UserAddressResource($this->userAddressRepository->create($data));

kira kira lebih enak dibaca yang mana ?

Reply:
enakan di baca yang kasih variable $data mas, karena dalam satu baris itu ada create 2 kali, jadi kalo di liat agak bingung juga ya, di create terus create lagi :-)